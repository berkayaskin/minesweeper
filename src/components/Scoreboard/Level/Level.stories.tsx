import { Meta, Story } from '@storybook/react/types-6-0'
import { Level, LevelProps } from './Level'

export default {
  title: 'Scoreboard/Level',
  component: Level,
} as Meta

const Template: Story<LevelProps> = (args) => <Level {...args} />

export const LevelExample = Template.bind({})
LevelExample.args = {
  children: ['Beginner', 'Intermediate', 'Expert'],
}
