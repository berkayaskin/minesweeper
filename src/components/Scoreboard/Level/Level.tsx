import styled from '@emotion/styled'
import { ChangeEvent } from 'react'

export interface LevelProps {
  /**
   * Array of possible game levels
   */
  children: string[]
  /**
   * Select new lvl handler
   */
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void
}

export const Level = ({ children, onChange }: LevelProps) => (
  <Select onChange={onChange}>
    {children.map((item: string) => (
      <Option key={item} value={item}>
        {item}
      </Option>
    ))}
  </Select>
)

const Select = styled.select`
  margin: 0;
  height: 100%;
  border-radius: 0;
  border: 0.15vw solid;
  border-color: white #9e9e9e #9e9e9e white;
  background-color: #d1d1d1;
`

const Option = styled.option`
  font-weight: normal;
  display: block;
  white-space: nowrap;
  min-height: 1.2em;
  padding: 0 0.2vw 0.2vw;
`
