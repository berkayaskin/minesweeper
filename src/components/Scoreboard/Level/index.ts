export { Level } from './Level'
export type { LevelProps } from './Level'
