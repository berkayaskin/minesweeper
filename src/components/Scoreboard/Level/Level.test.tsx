import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Level } from './Level'

describe('Level test cases', () => {
  it('Level renders correctly', () => {
    const onChange = jest.fn()

    const { asFragment } = render(
      <Level onChange={onChange}>{['beginner', 'intermediate', 'expert']}</Level>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
  it('Select level behaviour', () => {
    const onChange = jest.fn()

    render(<Level onChange={onChange}>{['beginner', 'intermediate', 'expert']}</Level>)

    userEvent.selectOptions(screen.getByRole('combobox'), 'intermediate')

    expect(screen.getByRole('option', { name: 'intermediate' })).toBeEnabled()

    expect(onChange).toHaveBeenCalled()
  })
})
