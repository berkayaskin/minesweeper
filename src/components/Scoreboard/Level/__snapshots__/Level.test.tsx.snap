// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level test cases Level renders correctly 1`] = `
<DocumentFragment>
  <select
    class="css-1lpmfg0"
  >
    <option
      class="css-zcmdxr"
      value="beginner"
    >
      beginner
    </option>
    <option
      class="css-zcmdxr"
      value="intermediate"
    >
      intermediate
    </option>
    <option
      class="css-zcmdxr"
      value="expert"
    >
      expert
    </option>
  </select>
</DocumentFragment>
`;
